platforms/posix/src/px4/common/main.cpp

```diff
@@ -174,6 +177,25 @@ int main(int argc, char **argv)
 		return client.process_args(argc, (const char **)argv);
 
 	} else {
+#if defined(__PX4_LINUX)	// Linux specific
+
+		// TODO: Check CAP_IPC_LOCK instead of uid
+		// try to lock address space into RAM, to avoid page swap delay
+		if (geteuid() != 0) {	// not the root user
+			PX4_WARN("mlockall() disabled. Take care of memory usage.");
+
+		} else {
+			if (mlockall(MCL_CURRENT) + mlockall(MCL_FUTURE)) {
```

Comment:
Any concern with `MCL_FUTURE` potentially causing future allocations to fail unexpectedly?

Reply:
None, I think. There is no limitation if the process is started by root. alloc() only fails if the RAM is all eaten by PX4, and, nothing we can do to avoid crashing(almost impossible). PX4 will crowd out other processes in RAM if it is full, and this action generally won't be the problem. We may not concern the startup memory allocation time, as we only require stable scheduling when the thread is fully started.